aspnetcore/mvc/views/tag-helpers/builtin/resources/CacheTagHelper.md

```diff
@@ -0,0 +1,286 @@
+
+- - -
+
+### vary-by
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| String             | "@Model"  	            |
+
+
+The attribute `vary-by` allows for customization of what data gets cached. When the object referenced by the attribute's string value changes, the content of the Cache Tag Helper is updated. Often a string-concatenation of model values are assigned to this attribute.  Effectively, that means any update to any of the concatenated values invalidates the cache.
+
+The example below assumes the controller method rendering the view sums the integer value of the two route parameters, `myParam1` and `myParam2`, and returns that as the single model property. When this sum changes, the content of the Cache Tag Helper is rendered again, and the cache is reset.  
+
+Usage Example:
+
+
+*Controller*
+
+```csharp
+public IActionResult Index(string myParam1,string myParam2,string myParam3)
+{
+    int num1;
+    int num2;
+    int.TryParse(myParam1, out num1);
+    int.TryParse(myParam2, out num2);
+    return View(viewName, num1 + num2);
+}
+```
+
+*Index.cshtml*
+
+```html
+<Cache vary-by="@Model"">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### priority
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| CacheItemPriority  | "High"   	            |
+|                    | "Low" |
+|                    | "NeverRemove" |
+|                    | "Normal" |
+
+The attribute `priority` provides cache ejection guidance to the built-in cache provider. If the web server experiences memory pressure and evicts one or more of your cached entities, it will evict Low priority cache entities first.
+
+Usage Example:
+
+```html
+<Cache priority="High">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+> [!WARNING]
+>  using the `priority` attribute does not guarantee any specific level of cache retention. It is simply a suggestion to the Cache Provider. To be extra clear, setting this attribute to `NeverRemove` does not guarantee that the cache will not be evicted.  More details can be found in the additional resources listed below.
+
+- - -
+
+
+>[!NOTE]
+> The Cache Tag Helper is dependent on the built in Memory Cache service being included in the `startup.cs` ConfigureServices method.  Adding this service is not necessary because the built in Tag Helper extensions take care of this in the `AddMvc` method call.
```

Comment:
* "built in" ... to "built-in"
* "service being included" ... to "service included"
* Capitalize "S" in "Startup.cs" and that filename should be in italics
* Put `ConfigureServices` in code with single backticks
* Two spaces after the first sentence